app/assets/javascripts/save-on-change.coffee

```diff
@@ -179,21 +179,29 @@ class @ForwardBlocker
     @binding.bind 'navigate_away', (e,opts) =>
       @navigate_away(opts.click_element, opts.action_to_perform)
 
-  prevent_forward_navigation_for: (blocker) ->
-    if blocker not in @blockers
-      @blockers.push blocker
+  get_blocker_index: (blocker) ->
+    firstBlocker = (b for b in @blockers when b.blocker is blocker)[0]
+    @blockers.indexOf(firstBlocker)
+
+  get_latest_disabled_forward_nav_message: ->
+    (b.disabledForwardNavMessage for b in @blockers when b.disabledForwardNavMessage?).reverse()[0]
```

Comment:
The name `disabledForwardNavMessage` is confusing to me. It sounds like a flag.  I think just `message` works here.  Assuming these objects are called Blockers then this is the message of the Blocker which makes sense to me.